mint-client/src/clients/user.rs

```diff
@@ -62,23 +67,32 @@ impl UserClient {
 
     fn ln_client(&self) -> LnClient {
         LnClient {
-            context: self.context.borrow_with_module_config(|cfg| &cfg.ln),
+            context: self.context.borrow_with_module_config(|cfg| &cfg.client.ln),
         }
     }
 
     fn mint_client(&self) -> MintClient {
         MintClient {
-            context: self.context.borrow_with_module_config(|cfg| &cfg.mint),
+            context: self
+                .context
+                .borrow_with_module_config(|cfg| &cfg.client.mint),
         }
     }
 
     fn wallet_client(&self) -> WalletClient {
         WalletClient {
-            context: self.context.borrow_with_module_config(|cfg| &cfg.wallet),
-            fee_consensus: self.context.config.fee_consensus.clone(), // TODO: remove or put into context
+            context: self
+                .context
+                .borrow_with_module_config(|cfg| &cfg.client.wallet),
+            fee_consensus: self.context.config.client.fee_consensus.clone(), // TODO: remove or put into context
         }
     }
 
+    // FIXME: we probably want a "gateway client"
+    pub fn gateway_api(&self) -> String {
+        self.context.config.gateway.api.clone()
+    }
```

Comment:
This is required for the POST to `{gateway_host}/pay_invoice`. Putting HTTP calls in `UserClient` doesn't seem right. Maybe a `GatewayClient` that can be accessed by the `UserClient` in the same way the other clients are?